### PR TITLE
Expand the /power api to support the ucs-obm service's power action

### PR DESCRIPTION
**Pull request by tannoa2**

### CODE REVIEW

**Comment by tannoa2** [comment]
@RackHD/veyron

File: controllers/ucs_controller.py [hunk 1]
```diff
@@ -268,11 +286,20 @@ def _service_profile_power_set(handle, dn=None, state=None):
         state = LsPowerConsts.STATE_CYCLE_IMMEDIATE
     elif state == "cycle-wait":
         state = LsPowerConsts.STATE_CYCLE_WAIT
+    elif state == "bmc-reset-immediate":
+        state = LsPowerConsts.STATE_CYCLE_WAIT
+    elif state == "ipmi-reset":
+        state = LsPowerConsts.STATE_CYCLE_WAIT
+    elif state == "hard-reset-immediate":
+        state = LsPowerConsts.STATE_CYCLE_WAIT
+    elif state == "soft-shut-downt":
```
Comment (geoff-reid) [inline controllers/ucs_controller.py]: typo: should be "soft-shut-down"

File: controllers/ucs_controller.py [hunk 2]
```diff
@@ -268,11 +286,20 @@ def _service_profile_power_set(handle, dn=None, state=None):
...
+        state = LsPowerConsts.STATE_CYCLE_WAIT
     else:
         raise UcsException(
             "server_power_set: Failed to set server power",
-            "action %s is not valid. Choose one of the following: "
-            "'on', 'off', 'cycle-wait'or 'cycle-immediate'".format(state))
+            "action '{0}' is not valid. Choose one of the following: "
+            "'on', 'off', 'cycle-wait','cycle-immediate', 'bmc-reset-immediate',"
+            " 'ipmi-reset', 'hard-reset-immediate', 'soft-shut-downt' ".format(state))
```
Comment (geoff-reid) [inline controllers/ucs_controller.py]: type: should be "soft-shut-down"

File: controllers/ucs_controller.py [hunk 3]
```diff
@@ -228,6 +228,24 @@ def getServiceProfile():
...
+    handle = UcsHandle(*authInfo, secure=False)
+    if handle.login():
+        try:
+            state = _powerStatus(identifier, handle)
+        except UcsException as e:
+            handle.logout()
+            return 'Internal Server Error', e.error_descr, 500
+        else:
+            handle.logout()
+        return "Server State is: " + state
```
Comment (geoff-reid) [inline controllers/ucs_controller.py]: It would be better to return a json object here instead of an English string.

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

File: controllers/ucs_controller.py [hunk 4]
```diff
@@ -228,6 +228,24 @@ def getServiceProfile():
 
 
 @http_body_factory
+def powerStatus(host=None, user=None, password=None, identifier=None):
```
Comment (keedya) [inline controllers/ucs_controller.py]: It looks like you are pulling the host/user from the header; Do you still need them here?

**Pull request merged**